nifti2/nifti_tool.c

```diff
@@ -3013,7 +3013,7 @@ int act_disp_hdr( nt_opts * opts )
    void        * nhdr;
    field_s     * fnhdr;
    const char ** sptr;
-   int           nfields, filenum, fc, nver=0;
+   int           nfields, filenum, fc, nver;
```

Comment:
I have never seen a similar initialization removed before.  It seems a bit unnecessary, but okay.

Reply:
This will have been:

`nifti2/nifti_tool.c:3007:44: warning: Variable 'nver' is assigned a value that is never used. [unreadVariable]`

It was assigned 0 at declaration, then unconditionally assigned a value a few lines later, thus making the initial assignment useless.

There was no real bug, but the warning is about this pattern:

1. x = something
2. do_something_with (x)
3. x = something

With no step 2 (as was the case here), it could often be a bug. (two assignments in a row with no intervening uses.)

Reply:
The logic is clear.  Initializing variables at definition generally seems like a good habit.  Now it is yet another thing that compilers will whine about.

Reply:
They only whine about it if the value at initialization is never read, and is overwritten by something else.  Like:

```c
int x = 3;
do_stuff_without_x();
x = 1
```

It's saying: why bother with the `= 3`?